New: Admin/Admin-top-bar - Basic usage. (ED-2797)

nisan250 opened the pull request:
## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

nisan250 committed:
ED-2797: Add Top Bar to admin pages

nisan250 committed:
ED-2797 - Add elementor top bar

Now we have a basic elementor top bar which appear in elementor pages with basic functionality for finder and connect.

nisan250 committed:
adding hover state

nisan250 committed:
Fix lint issues

nisan250 committed:
Adding the base for rendering title and buttons

nisan250 committed:
Adding more functionalities for the admin top bar

matipojo commented on core/common/modules/connect/module.php:
```diff
@@ -198,4 +198,16 @@ public function get_subscription_plans( $context ) {
 		];
 	}
 
+	protected function get_init_settings() {
```
@Nevoss it's something you changed in another PR?

Nevoss replied:
@matipojo no, @nisan250 needs to know if the user is connected or no to show the "connect" button in the top bar, although it can be useful for the kit library.

matipojo commented on core/modules-manager.php:
```diff
@@ -92,6 +92,8 @@ public function get_modules_names() {
 			'compatibility-tag',
 			'elements-color-picker',
 			'shapes',
+			// Must be the last module in the list.
```
Explain why

nisan250 replied:
Removed

matipojo commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -0,0 +1,60 @@
+import BarHeading from 'elementor/modules/admin-top-bar/assets/js/components/bar-heading/bar-heading';
```
```suggestion
import BarHeading from './components/bar-heading/bar-heading';
```

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -0,0 +1,60 @@
+export default function AdminTopBar() {
+	const actionButtonsRef = useRef();
+	const [ pageTitle, setPageTitle ] = useState( null );
+	// const [ actionButtons, setActionButtons ] = useState( null );
```
Remove

nisan250 replied:
Done

matipojo commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -0,0 +1,60 @@
+	useEffect( () => {
+		const pageTitleElement = document.querySelector( '.wp-heading-inline' );
+		if ( pageTitleElement ) {
+			const pageTitleText = pageTitleElement.innerText;
+			pageTitleElement.remove();
+			setPageTitle( pageTitleText );
```
```suggestion
			setPageTitle( pageTitleElement.innerText );
			pageTitleElement.remove();
```

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -0,0 +1,60 @@
+		$e.route( 'finder' );
+	};
+
+	const renderConnectButton = () => {
```
```suggestion
	const ConnectButton = () => {
```

matipojo commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -0,0 +1,60 @@
+			</div>
+
+			<div className="bar-action-area bar-flex">
+				<BarButton onClick={finderAction} icon="eicon-search-bold">Finder</BarButton>
```
Use BarFinder? (+Translate)

nisan250 replied:
Translated.
I am not sure about the BarFinder, This is a reusable component which I can use not only for finder but for the account button.

matipojo commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -0,0 +1,60 @@
+		}
+	}, [] );
+
+	const isUserConnected = elementorCommonConfig.connect.is_user_connected;
+	let connectUrl = '';
```
Move to the component.

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -0,0 +1,60 @@
+
+		if ( isUserConnected ) {
+			connectUrl = 'https://go.elementor.com/wp-dash-admin-bar-account/';
+			return <BarButton icon="eicon-user-circle-o" onClick={connectAction}>My Elementor</BarButton>;
+		}
+		connectUrl = elementorCommonConfig.connect.connect_url;
+		return <BarButton icon="eicon-user-circle-o" onClick={connectAction}>Connect Account</BarButton>;
```
Translate

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/assets/js/components/bar-heading/bar-heading.js:
```diff
@@ -0,0 +1,14 @@
+export default function BarHeading( props ) {
+	const logoSource = `${ elementorCommon.config.urls.assets }images/logo-icon-rounded.png`;
```
Ask for an SVG file.

nisan250 replied:
I have changed it and now I use the e-icon class like in "theme builder".

matipojo commented on modules/admin-top-bar/assets/scss/_index.scss:
```diff
@@ -0,0 +1,119 @@
+$top-bar-height: 50px;
+
+#e-admin-top-bar {
+	background: #fff;
+	box-shadow: 0px 4px 6px rgba(0, 0, 0, 0.0394871);
+	position: absolute;
+	top: 0;
+	right: 0;
+	width: calc(100% - 160px);
+	z-index: 9999;
```
Don't use value like `50px` `9999`, `#fff`. use pre-defined units and variables.

nisan250 replied:
I have changed the colors to colors that exists in our variables file and used the z-index variable but I am not sure about the 50px, this values are not pre defined anywhere at least I am not aware of.

matipojo replied:
It's a unit from the Design System, not a variable. @joshmarom can you advise?

nisan250 replied:
Josh said I cant use it as a framework, it was something very specific to a feature he worked on.

matipojo commented on .grunt-config/sass.js:
```diff
@@ -4,13 +4,22 @@ const sass = {
+				dest: 'assets/css',
+				ext: '.css'
+			},
+			{
+				expand: true,
+				cwd: 'modules/admin-top-bar/assets/scss',
+				src: '*.scss',
+				dest: 'assets/css/modules/admin-top-bar',
+				ext: '.css',
+			},
```
Remove. add the needed file to the `direction` folder,

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/module.php:
```diff
@@ -0,0 +1,130 @@
+}
+
+class Module extends BaseApp {
+	//private $elementor_pages = [];
```
Remove.

nisan250 replied:
Done

matipojo commented on modules/admin-top-bar/module.php:
```diff
@@ -0,0 +1,130 @@
+
+class Module extends BaseApp {
+	//private $elementor_pages = [];
+	const ELEMENTOR_MENU_ITEM_IDS = [
+		'toplevel_page_elementor',
+		'menu-posts-elementor_library'
+
+	];
```
Why do you need it on the server side?

nisan250 replied:
When a developer will add new page, it will be more easy for him to see straight in the module the classes which the top bar can be added, and if he need to add new one, he will do it in the module.

matipojo replied:
But it's not related to the server, so, please move it to the main App.

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/module.php:
```diff
@@ -0,0 +1,130 @@
+	 * @return bool
+	 */
+	public static function is_active() {
+		return is_admin() && static::is_elementor_page();
```
```suggestion
		return is_admin();
```

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/module.php:
```diff
@@ -0,0 +1,130 @@
+	 */
+	public static function is_active() {
+		return is_admin() && static::is_elementor_page();
+		//return ! _is_elementor_installed();
```
Remove.

nisan250 replied:
Done

matipojo commented on modules/admin-top-bar/module.php:
```diff
@@ -0,0 +1,130 @@
+
+		$this->print_config();
+
+		//wp_set_script_translations(
+		//	'form-submission-admin',
+		//	'elementor',
+		//	ELEMENTOR_PRO_PATH . 'languages'
+		//);
```
Remove all commented code.

nisan250 replied:
Done.

nisan250 commented on modules/admin-top-bar/assets/js/admin.js:
```diff
@@ -0,0 +1,17 @@
+
+const AppWrapper = elementorCommon.config.isDebug ? React.StrictMode : React.Fragment;
+
+const menuItemClasString = elementorAdminTopBarConfig.elementor_menu_item_ids
```
Fixed

nisan250 commented on modules/admin-top-bar/assets/js/admin.js:
```diff
@@ -0,0 +1,17 @@
+const AppWrapper = elementorCommon.config.isDebug ? React.StrictMode : React.Fragment;
+
+const menuItemClasString = elementorAdminTopBarConfig.elementor_menu_item_ids
+	.map( ( elementorMenuItemId ) => `#${ elementorMenuItemId } .wp-menu-open` )
```
Done.

nisan250 committed:
Review fixes

matipojo commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -0,0 +1,59 @@
+import BarHeading from './components/bar-heading/bar-heading';
+import BarButton from './components/bar-button/bar-button';
```
a-z

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/module.php:
```diff
@@ -0,0 +1,98 @@
+		];
+	}
+
+	private static function is_elementor_page() {
+		return true;
+	}
```
?

nisan250 replied:
Removed.

matipojo commented on modules/admin-top-bar/module.php:
```diff
@@ -0,0 +1,98 @@
+
+		wp_enqueue_script(
+			'elementor-admin-top-bar',
+			$this->get_js_assets_url( 'admin-top-bar' ),   // create path in directory tree
+			[
+				'react',     //dependencies
```
The comments are unnecessary.

nisan250 replied:
Removed.

nisan250 committed:
Adding mobile mode according to the guidelines

matipojo commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -2,9 +2,25 @@ import BarHeading from './components/bar-heading/bar-heading';
 export default function AdminTopBar() {
 	const actionButtonsRef = useRef();
 	const [ pageTitle, setPageTitle ] = useState( null );
+	const [ isMobile, setIsMobile ] = useState( getIsMobile() );
+
+	useEffect( () => {
+		function handleResize() {
+			setIsMobile( getIsMobile() );
+		}
+
+		window.addEventListener( 'resize', handleResize );
+		return () => window.removeEventListener( 'resize', handleResize );
+	}, [] );
```
Use CSS & a media query.

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -35,24 +51,22 @@ export default function AdminTopBar() {
-		}
-		connectUrl = elementorCommonConfig.connect.connect_url;
-		return <BarButton icon="eicon-user-circle-o" onClick={connectAction}>{ __( 'Connect Account', 'elementor' ) }</BarButton>;
+		connectUrl = isUserConnected ? 'https://go.elementor.com/wp-dash-admin-bar-account/' : elementorCommonConfig.connect.connect_url;
```
```suggestion
		const connectUrl = isUserConnected ? 'https://go.elementor.com/wp-dash-admin-bar-account/' : elementorCommonConfig.connect.connect_url;
```

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -35,24 +51,22 @@ export default function AdminTopBar() {
-		connectUrl = elementorCommonConfig.connect.connect_url;
-		return <BarButton icon="eicon-user-circle-o" onClick={connectAction}>{ __( 'Connect Account', 'elementor' ) }</BarButton>;
+		connectUrl = isUserConnected ? 'https://go.elementor.com/wp-dash-admin-bar-account/' : elementorCommonConfig.connect.connect_url;
+		const buttonText = isUserConnected ? __( 'My Elementor', 'elementor' ) : __( 'Connect Account', 'elementor' );
```
Combine consts

nisan250 replied:
Done.

nisan250 committed:
Handle page layout when admin top bar is visible/invisible

nisan250 committed:
Refactor mobile support 

In mobile mode the action buttons will be shown i admin top bar, in desktop mode the action buttons will be shown i admin page

nisan250 committed:
Review fixes

nisan250 committed:
Remove comment

nisan250 committed:
Adding the ability to inject elements from the cloud

nisan250 committed:
Adding default page title when not exist

nisan250 committed:
Add tooltip to connect account

nisan250 committed:
Fixing the import templates button to work with admin top bar

nisan250 committed:
Support RTL

nisan250 commented on modules/admin-top-bar/assets/scss/_index.scss:
```diff
@@ -0,0 +1,133 @@
+@import "../../../../assets/dev/scss/helpers/direction";
+@import "../../../../assets/dev/scss/helpers/variables";
+@import "../../../../assets/dev/scss/helpers/mixins";
+$top-bar-height: 50px;
```
Josh said I cant use it as a framework, it was something very specific to a feature he worked on.

nisan250 committed:
Fixing the display of the action buttons

nisan250 committed:
Refactor the loading css assets

nisan250 committed:
Moving the logic for the connection button to a new component

Nevoss commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -0,0 +1,60 @@
+	const [ pageTitle, setPageTitle ] = useState( null );
+
+	// Indicate that the admin top bar is visible and the page content needs to push down below the admin top bar for visibility.
+	useEffect( () => {
+		const adminTopBarElement = document.querySelector( '#e-admin-top-bar' );
+		adminTopBarElement.classList.add( 'top-bar-active' );
+	}, [] );
```
just suggestion not required: 

```suggestion
	useEffect( () => {
		document.querySelector( '#e-admin-top-bar' );
			classList.add( 'top-bar-active' );
	}, [] );
```

nisan250 replied:
I will it as is.

Nevoss commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -0,0 +1,60 @@
+export default function AdminTopBar() {
+	let actionButtonElements;
+	const actionButtonsRef = useRef();
+	const [ pageTitle, setPageTitle ] = useState( null );
```
```suggestion
	const [ pageTitle, setPageTitle ] = useState( 'Elementor' );
```

nisan250 replied:
Done.

Nevoss commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -0,0 +1,60 @@
+	}, [] );
+
+	// Handle the page title visibility in admin top bar.
+	useEffect( () => {
+		const pageTitleElement = document.querySelector( '.wp-heading-inline' );
+		if ( pageTitleElement ) {
+			const pageTitleText = pageTitleElement;
+			pageTitleElement.remove();
+			setPageTitle( pageTitleText.innerText );
+		} else {
+			setPageTitle( 'Elementor' );
+		}
+	}, [] );
```
Can be external hook `const pageTitle = usePageTitle()`

```suggestion
	useEffect( () => {
		const pageTitleElement = document.querySelector( '.wp-heading-inline' );
		if ( ! pageTitleElement ) {
			return;
		}
		
		const pageTitleText = pageTitleElement;
		pageTitleElement.remove();
		setPageTitle( pageTitleText.innerText );
	}, [] );
```

nisan250 replied:
Done.

Nevoss commented on modules/admin-top-bar/assets/js/components/page-button/page-button.js:
```diff
@@ -0,0 +1,9 @@
+export default function pageButton( props ) {
```
Is this component in use?

nisan250 replied:
Not anymore. removed.

Nevoss commented on modules/admin-top-bar/assets/scss/admin-rtl.scss:
```diff
@@ -0,0 +1,3 @@
+$direction: rtl;
```
is it in use?

nisan250 replied:
Yes, I use @include end( 0 );

matipojo replied:
Remove the file

nisan250 replied:
Done.

Nevoss commented on modules/admin-top-bar/assets/scss/admin.scss:
```diff
@@ -0,0 +1,3 @@
+$direction: ltr;
```
is it in use?

nisan250 replied:
Yes, I use @include end( 0 );

Nevoss commented on modules/admin-top-bar/module.php:
```diff
@@ -0,0 +1,100 @@
+
+	private function render_admin_top_bar() {
+		?>
+			<div id="e-admin-top-bar">
```
format

nisan250 replied:
Done.

nisan250 commented on modules/admin-top-bar/assets/js/admin.js:
```diff
@@ -0,0 +1,23 @@
+const elementorMenuItemIds = [
+	'toplevel_page_elementor',
+	'menu-posts-elementor_library',
+
```
Done.

nisan250 commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState, useRef } from 'react';
+
+export default function AdminTopBar() {
+	let actionButtonElements;
```
Not anymore. removed.

nisan250 commented on .grunt-config/sass.js:
```diff
@@ -4,13 +4,15 @@ const sass = {
-			dest: 'assets/css',
-			ext: '.css'
-		} ]
+		files: [
```
Done.

nisan250 commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -0,0 +1,60 @@
+import BarButton from './components/bar-button/bar-button';
```
Done.

nisan250 commented on modules/admin-top-bar/assets/js/app.js:
```diff
@@ -0,0 +1,60 @@
+
+	// Handle the action buttons visibility in admin top bar on initiation.
+	useEffect( () => {
+		actionButtonElements = document.querySelectorAll( '.page-title-action' );
+		if ( actionButtonElements ) {
+			actionButtonElements.forEach( ( actionButtonElement ) => {
+				actionButtonsRef.current.appendChild( actionButtonElement );
+			} );
+		}
```
Removed if statement.

nisan250 committed:
Review fixes

nisan250 committed:
Creating custom hook for page title

Nevoss commented on modules/admin-top-bar/module.php:
```diff
@@ -0,0 +1,85 @@
+
+		wp_enqueue_style( 'elementor-admin-top-bar' );
+
+		wp_enqueue_script( 'elementor-admin-top-bar', $this->get_js_assets_url( 'admin-top-bar' ), [
+				'react',
+				'react-dom',
+			], ELEMENTOR_VERSION, true );
+
+		$min_suffix = Utils::is_script_debug() ? '' : '.min';
+
+		wp_enqueue_script( 'tipsy', ELEMENTOR_ASSETS_URL . 'lib/tipsy/tipsy' . $min_suffix . '.js', [
+				'jquery',
+			], '1.0.0', true );
```
Something wrong with the indentation here

nisan250 replied:
Done.

nisan250 committed:
Review fixes

nisan250 committed:
Add administrator permissions

matipojo commented on assets/dev/scss/admin/admin-top-bar.scss:
```diff
@@ -0,0 +1,173 @@
+@import "../../../../assets/dev/scss/helpers/direction";
```
@joshmarom can you review this file?

matipojo commented on modules/admin-top-bar/assets/js/admin-top-bar.js:
```diff
@@ -0,0 +1,48 @@
+		<div id="elementor-admin-top-bar">
+			<div className="bar-main-area bar-flex">
+				<BarHeading>{ pageTitleText }</BarHeading>
+				<div className="top-bar-action-buttons-wrapper" ref={ actionButtonsRef } />
```
Use BEM class names convention. consult with @joshmarom

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/assets/scss/_index.scss:
```diff
@@ -0,0 +1,173 @@
+@import "../../../../assets/dev/scss/helpers/direction";
```
Move to `modules/admin-top-bar/assets/scss/admin.scss`

nisan250 replied:
Done.

matipojo commented on assets/dev/scss/admin/admin-top-bar.scss:
```diff
@@ -0,0 +1,173 @@
+@import "../../../../assets/dev/scss/helpers/mixins";
+@import "../../../../assets/dev/scss/editor/breakpoints";
+
+$top-bar-height: 50px;
```
Why is the code here?
you should include the file at `modules/admin-top-bar/assets/scss/admin.scss`

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/module.php:
```diff
@@ -0,0 +1,90 @@
+	private function enqueue_scripts() {
+		$direction_suffix = is_rtl() ? '-rtl' : '';
+
+		wp_register_style( 'elementor-admin-top-bar', $this->get_css_assets_url( 'admin-top-bar' . $direction_suffix ), [], ELEMENTOR_VERSION );
```
`get_css_assets_url` is already handles the `rtl` asspect

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/assets/js/components/connection-button/connection-button.js:
```diff
@@ -0,0 +1,23 @@
+import BarButton from '../bar-button/bar-button';
+
+export default function ConnectionButton( props ) {
+	const isUserConnected = elementorCommonConfig.connect.is_user_connected;
```
Use the elementorConnect plugin like on 
https://github.com/matipojo/elementor/blob/e86b957a2ebdf75a4c4c74c18f8e6adf33a3df99/assets/dev/js/editor/components/template-library/views/parts/connect.js#L16

nisan250 replied:
Done.

nisan250 committed:
Add admin top bar to the dashboard

nisan250 committed:
Review fixes

nisan250 committed:
Using BEM

matipojo commented on modules/admin-top-bar/assets/js/admin-top-bar.js:
```diff
@@ -0,0 +1,66 @@
+	};
+
+	return (
+		<div id="elementor-admin-top-bar">
```
Replace all `elementor` with `e`, e.g. `e-admin-top-bar`

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/assets/js/hooks/use-page-title/use-page-title.js:
```diff
@@ -0,0 +1,21 @@
+		}
+
+		if ( isDashboard ) {
+			setPageTitle( 'Dashboard' );
```
Translate

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/assets/scss/admin.scss:
```diff
@@ -0,0 +1,211 @@
+
+// Hide the widget in admin dashboard that was created by checking the screen option checkbox.
+// "display: none" won't work here since it will influence the mechanism for saving the state for the checkbox in screen options
+#e-dashboard-widget-admin-top-bar {
```
Just use it instead of your React root. By using it you also don't need to deal with the checkbox changes in `AdminTopBar` component

nisan250 replied:
@matipojo To use the widget as the container of admin top bar, I will have to give it a position absolute, but this wont take it to the top of the page since there is a positioned ancestor. giving it a fixed position wont give us the necessary behavior as describe in the task.

nisan250 replied:
After a discussion we decided not to do it.

matipojo commented on modules/admin-top-bar/assets/scss/admin.scss:
```diff
@@ -0,0 +1,211 @@
+		}
+	}
+
+	&:not(.top-bar-active )  ~ #wpbody {
```
```suggestion
	&:not(.top-bar-active ) ~ #wpbody {
```

nisan250 replied:
Done.

nisan250 committed:
Review fixes

nisan250 committed:
Extend the connect button to work in admin

matipojo commented on modules/admin-top-bar/assets/scss/admin.scss:
```diff
@@ -0,0 +1,211 @@
+	pointer-events: none;
+}
+
+#e-admin-top-bar {
```
```suggestion
#e-admin-top-bar-root {
```

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/assets/scss/admin.scss:
```diff
@@ -0,0 +1,211 @@
+	display: none;
+	position: absolute;
+	top: 0;
+	width: calc(100% - 160px);
```
Explain the 160px

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/assets/scss/admin.scss:
```diff
@@ -0,0 +1,211 @@
+	.e-admin-top-bar__bar-button-title {
+		font-size: 13px;
+		font-weight: 500;
+		font-family: 'Roboto';
```
Where is the font enqueue? How should it work for other languages?

nisan250 replied:
Done.I have added the fonts in enqueue_scripts.

matipojo commented on modules/admin-top-bar/assets/scss/admin.scss:
```diff
@@ -0,0 +1,211 @@
+
+@media screen and (max-width: 960px) {
+	#e-admin-top-bar {
+		width: calc(100% - 36px);
```
Explain the 36px

nisan250 replied:
Done.

matipojo commented on modules/admin-top-bar/assets/scss/admin.scss:
```diff
@@ -0,0 +1,211 @@
+@media (max-width: $editor-screen-md-min) {
+	#e-admin-top-bar {
+		.e-admin-top-bar__main-area-buttons {
+			position: absolute;
+			top: calc(100% + 10px);
+		}			display: inline-flex;
```
Fix identation

nisan250 replied:
Done

nisan250 committed:
Review fixes

nisan250 committed:
Enqueue roboto google font

nisan250 commented on modules/admin-top-bar/assets/js/admin.js:
```diff
@@ -0,0 +1,25 @@
+	if ( isElementorPage || isTopBarOptionWidgetChecked ) {
+		ReactDOM.render(
+			<AppWrapper>
+				<AdminTopBar isDashboard={isTopBarOptionWidgetChecked} />
```
There is leftover code in JS and maybe CSS. after talking with Aviad, he decided to open another task for that and merge this code as fast as we can.
